module/buffer.go

```diff
@@ -3,19 +3,18 @@ package module
 import (
 	"github.com/onflow/flow-go/model/cluster"
 	"github.com/onflow/flow-go/model/flow"
-	"github.com/onflow/flow-go/model/messages"
 )
 
 // PendingBlockBuffer defines an interface for a cache of pending blocks that
 // cannot yet be processed because they do not connect to the rest of the chain
 // state. They are indexed by parent ID to enable processing all of a parent's
 // children once the parent is received.
 type PendingBlockBuffer interface {
```

Comment:
We could also make this interface generic to work both with cluster blocks?

Reply:
I have spent some time experimenting  with this - we can, but the exact approach has impacts beyond this module. I would prefer to spend some more time experimenting and choosing the right approach before merging anything. I've pushed all the WIP code to `jordan/3374--generics-experimentation-mess`.